### PR TITLE
Initial draft

### PR DESCRIPTION
**Pull request by htrivedi99**

### CODE REVIEW

File: blog/2023-08-10-evaluating-open-source-llms/index.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+Person 3: Maybe we should try MPT! It's known for its amazing fluency and coherence in generated text.
+Person 1: Yeah, but wait, Falcon looks cool too. It claims to handle a wide range of language tasks effortlessly.
+Person 2: And LLama is available for commercial use, so that's something to consider.
+```
+
+## How LLMs Are Benchmarked
+
+Most people that play around with LLMs can tell how well a model performs just by the output they're getting. But how can you numerically measure an LLM's performance?
+
+Currently, the way LLMs are benchmarked is by testing them on a variety of datasets. The [hugging face leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard) has a nice visual representation of how well open-source LLMs perform on 4 standard datasets and where they rank in the leaderboard.
```
Comment (casperdcl) [inline blog/2023-08-10-evaluating-open-source-llms/index.md:35]: link looks borked?

image

File: blog/2023-08-10-evaluating-open-source-llms/index.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+Often times the numbers that you see from benchmarks don't seem to translate into real-world performance. For example, there might be a benchmark that shows that the Llama 70B model is superior to ChatGPT at some particular task. In reality, ChatGPT might be better even though the benchmark says otherwise.
+
+The reason is that the datasets used for these benchmarks are limited and do not cover all of the possible inputs an LLM could receive. Truth is, models developed by OpenAI, Cohere, and Anthropic are way larger in size (100B + parameters). On top of that, they are trained on way more data compared to their open-source counterparts. 
+
+So, the key takeaway is to use benchmarks as a starting point for evaluating LLMs, but not rely on them entirely.
+
+
+## Reinforcement Learning
+
+Open-source models don't have the same size as ChatGPT or GPT-4, but even the larger 70B parameter models can have issues related to bias and fairness. When GPT-3 was first released it has racial biases, gender biases, and religious biases. The origin of these biases was the data these models were trained on.
```
Comment (casperdcl) [inline blog/2023-08-10-evaluating-open-source-llms/index.md:118]: citation needed :)

File: blog/2023-08-10-evaluating-open-source-llms/banner.jpeg
Comment (casperdcl) [inline blog/2023-08-10-evaluating-open-source-llms/banner.jpeg:1]: where's this banner from? stable diffusion?

**Review by casperdcl — CHANGES_REQUESTED** [review]

File: blog/2023-08-10-evaluating-open-source-llms/index.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+By now, it seems like every month a new open-source Large Language Model(LLM) comes along and breaks all of the records held by previous models. 
+
+The pace at which generative AI is progressing is so quick that people are spending most of their time catching up rather than building useful tools. There is a lot of confusion in the open-source community as to which LLM is the best. Businesses want to use the best open-source LLM for their use case. But how do you know which one is the best?
+
+```txt
+Person 1: You know, I've been thinking, we should definitely consider using one of these large language models for our project.
+Person 2: True, but which one should we go for?
+Person 3: Maybe we should try MPT! It's known for its amazing fluency and coherence in generated text.
+Person 1: Yeah, but wait, Falcon looks cool too. It claims to handle a wide range of language tasks effortlessly.
+Person 2: And LLama is available for commercial use, so that's something to consider.
```
[the remark this answers is not in the document]
Comment (htrivedi99) [reply blog/2023-08-10-evaluating-open-source-llms/index.md]: fixed

File: blog/2023-08-10-evaluating-open-source-llms/index.md (hunk 1 above)
Comment (htrivedi99) [reply blog/2023-08-10-evaluating-open-source-llms/index.md:38]: I think the link should work now. Maybe the hugging face leaderboard was down when you tried it?

File: blog/2023-08-10-evaluating-open-source-llms/index.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+One thing to note is that this is just a subset of the benchmarks used to evaluate LLMs. These benchmarks are used by Hugging Face to evaluate all of the LLMs on their platform.
+
+## Benchmark Limitations
+
+Often times the numbers that you see from benchmarks don't seem to translate into real-world performance. For example, there might be a benchmark that shows that the Llama 70B model is superior to ChatGPT at some particular task. In reality, ChatGPT might be better even though the benchmark says otherwise.
+
+The reason is that the datasets used for these benchmarks are limited and do not cover all of the possible inputs an LLM could receive. Truth is, models developed by OpenAI, Cohere, and Anthropic are way larger in size (100B + parameters). On top of that, they are trained on way more data compared to their open-source counterparts. 
+
+So, the key takeaway is to use benchmarks as a starting point for evaluating LLMs, but not rely on them entirely.
```
[the remark this answers is not in the document]
Comment (htrivedi99) [reply blog/2023-08-10-evaluating-open-source-llms/index.md]: fixed

File: blog/2023-08-10-evaluating-open-source-llms/index.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+
+## Reinforcement Learning
+
+Open-source models don't have the same size as ChatGPT or GPT-4, but even the larger 70B parameter models can have issues related to bias and fairness. When GPT-3 was first released it has racial biases, gender biases, and religious biases. The origin of these biases was the data these models were trained on.
+
+It's nearly impossible to remove all biases from a dataset, but at the same time, we don't want the models to pick up on them. How do we fix this problem?
+
+This is where *Reinforcement Learning with Human Feedback(RLHF)* comes in. With RLHF, a human ranks the outputs of an LLM from best to worst. Each time the human ranks the outputs, they are essentially training a reinforcement model. This reinforcement model is then used to "reward" the model when it generates an output that is "good" and penalizes the model when the output is "bad".
```
[the remark this answers is not in the document]
Comment (htrivedi99) [reply blog/2023-08-10-evaluating-open-source-llms/index.md]: citation added

File: blog/2023-08-10-evaluating-open-source-llms/index.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+Using RLHF, the hidden biases a model has picked up from the training dataset can be removed, hence improving the model's accuracy.
+
+## Picking the right LLM
+
+Even though the LLMs on Hugging Face are benchmarked on the same datasets, each LLM excels at a particular task. To put it simply, even though Llama-2 currently dominates the open-source leaderboard, it does not mean that it's the best open-source LLM.
+
+The LLM you pick should depend on the type of problem you are solving. If you're trying to generate code to make API calls, maybe you want to use Gorilla. If you want to design a conversational chatbot, maybe try one of the Falcon models. Each LLM has its own advantages and disadvantages, and only through experimentation can you begin to understand which LLM is right for your use case.
```
[the remark this answers is not in the document]
Comment (htrivedi99) [reply blog/2023-08-10-evaluating-open-source-llms/index.md]: added links to prem registry

File: blog/2023-08-10-evaluating-open-source-llms/index.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+## How LLMs Are Benchmarked
+
+Most people that play around with LLMs can tell how well a model performs just by the output they're getting. But how can you numerically measure an LLM's performance?
+
+Currently, the way LLMs are benchmarked is by testing them on a variety of datasets. The [hugging face leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard) has a nice visual representation of how well open-source LLMs perform on 4 standard datasets and where they rank in the leaderboard. 
+
+Whenever a new LLM is open-sourced, it is tested on prompts from datasets like *ARC*, *HellaSwag*, *MMLU*, and *TruthfulQA*. The output of the LLM is compared with the ground truth value which is what is actually expected. For example, the MMLU dataset contains a question with multiple-choice answers.
+
+![](./diagram_1.jpg)
```
[the remark this answers is not in the document]
Comment (htrivedi99) [reply blog/2023-08-10-evaluating-open-source-llms/index.md:42]: This is the full picture. I thought the whole point of evaluating these LLMs was to test them with zero context. The only context they have is whatever data they are trained on. Aside from that, no context is provided for these benchmarks.

File: blog/2023-08-10-evaluating-open-source-llms/index.md (hunk 1 above)
Comment (casperdcl) [reply blog/2023-08-10-evaluating-open-source-llms/index.md:36]: yes working now. Heisenbug :)

File: blog/2023-08-10-evaluating-open-source-llms/index.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (casperdcl) [reply blog/2023-08-10-evaluating-open-source-llms/index.md:40]: Hmm I'm not sure about the value of testing LLMs without context. That seems to be only reasonable for AGI, not AI... and it also means that an objectively bad LLM will still do well if it happened to be trained on data similar to the task you happened to give it.

"testing an LLM without context" seems to me to be identical to "testing my ability to come up with tasks that weren't in the LLM's training data"... not really useful, right?

/CC @filopedraz

File: blog/2023-08-10-evaluating-open-source-llms/index.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+This is where [*Reinforcement Learning with Human Feedback(RLHF)*](https://wandb.ai/ayush-thakur/RLHF/reports/Understanding-Reinforcement-Learning-from-Human-Feedback-RLHF-Part-1--VmlldzoyODk5MTIx) comes in. With RLHF, a human ranks the outputs of an LLM from best to worst. Each time the human ranks the outputs, they are essentially training a reinforcement model. This reinforcement model is then used to "reward" the model when it generates an output that is "good" and penalizes the model when the output is "bad".
+
+Using RLHF, the hidden biases a model has picked up from the training dataset can be removed, hence improving the model's accuracy.
+
+Pros:
+- Increased efficiency: Using RLHF, the feedback that is provided helps guide the LLM toward the optimal solution. With only a few examples, a fine-tuned model with RLHF can easily outperform the baseline model on certain tasks.
+- Better Performance: The feedback that a human provides also impacts the quality of the output generated by an LLM. By showing more examples of the desired outcomes, the LLM improves the generated output to match what is expected.
```
Comment (casperdcl) [inline blog/2023-08-10-evaluating-open-source-llms/index.md:131]: these explanations don't make it clear what the difference between "efficiency" and "performance" is. They read like nearly identical bullet points. Do you actually mean "accuracy" and "quality"?

File: blog/2023-08-10-evaluating-open-source-llms/index.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+Cons:
+- Lack of scalability: RLHF depends on human feedback to improve the performance of the model. So, in this case, the human is the bottleneck. Providing feedback to an LLM can be a time-consuming process and it can't be automated. Because of this, RLHF is considered a slow and tedious process.
+- Inconsistent quality: Different people may be providing feedback for the same model and those people may have differing opinions on what should be the desired output. People make decisions based on their knowledge and preference, but too many differing opinions can confuse the model and lead to performance degradation.
```
Comment (casperdcl) [inline blog/2023-08-10-evaluating-open-source-llms/index.md:135]: would love a citation for "too many differing opinions can confuse the model and lead to performance degradation"

**Pull request merged**